### PR TITLE
Documentation PR: Guide on "Start Over" & "Task List" Options available with in Backstage Software template

### LINKED ISSUE #23966: 🚀 Feature: Documentation Request: Guide on "Start Over" & Previewing Previous Template Tasks in Backstage
### 🔖 Feature description

I would like to request documentation on how to use the "Start Over" and "Preview Previous Template Tasks" features within the Backstage portal. 

### 🎤 Context

As a user of the Backstage portal, I often find myself needing to start over an already executed task or preview previous template tasks. However, I couldn't find any documentation on how to do these. Having a guide would be very helpful and improve the user experience.

### ✌️ Possible Implementation

The documentation should include:

1. A step-by-step guide on how to start over an already executed task.
2. Instructions on how to preview previous template tasks.
3. Screenshots or GIFs demonstrating the process.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by subbu-raj**
## Hey, I just made a Pull Request!

Adding the documentation for "Task List" & "Start Over" Options available under Software Templates. This documentation can be viewed by navigating to Docs -> Core Features -> Software Templates -> Overview in "backstage.io"
fixes #23966

**Snap of the changes are below:**
Screenshot 2024-04-10 185438
Screenshot 2024-04-10 185509

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. 
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/software-templates/index.md [hunk 1]
```diff
@@ -103,3 +103,17 @@ from `backstage/packages/app/src/App.tsx`:
 ```
 
 After the change, you should no longer see the button.
+
+## Previewing and Executing Previous Template Tasks
+
+Each execution of a template is treated as a unique task, identifiable by its own unique ID. To view a list of previously executed template tasks, navigate to the `Create` page and access the `Task List` from the context menu (represented by the vertical ellipsis, or 'kebab menu', icon in the upper right corner).
```
Comment (awanlin) [inline docs/features/software-templates/index.md:109]:
```suggestion
Each execution of a template is treated as a unique task, identifiable by its own unique ID. To view a list of previously executed template tasks, navigate to the "Create" page and access the "Task List" from the context menu (represented by the vertical ellipsis, or 'kebab menu', icon in the upper right corner).
```

File: docs/features/software-templates/index.md [hunk 2]
```diff
@@ -103,3 +103,17 @@ from `backstage/packages/app/src/App.tsx`:
...
+
+![Template Task List](../../assets/software-templates/template-task-list.png)
+
+If you wish to re-run a previously executed template, navigate to the template tasks page. Locate the desired task and select the `Start Over` option from the context menu.
```
Comment (awanlin) [inline docs/features/software-templates/index.md:113]:
```suggestion
If you wish to re-run a previously executed template, navigate to the template tasks page. Locate the desired task and select the "Start Over" option from the context menu.
```

File: docs/features/software-templates/index.md [hunk 3]
```diff
@@ -103,3 +103,17 @@ from `backstage/packages/app/src/App.tsx`:
...
+
+![Template Start Over](../../assets/software-templates/template-start-over.png)
+
+This action will initiate a new execution of the selected template, pre-populated with the same parameters as the previous run, but these parameters can be edited before re-execution.
+
+In the event of a failed template execution, the `Start Over` option can be used to re-execute the template. The parameters from the original run will be pre-filled, but they can be adjusted as needed before retrying the template.
```
Comment (awanlin) [inline docs/features/software-templates/index.md:119]:
```suggestion
In the event of a failed template execution, the "Start Over" option can be used to re-execute the template. The parameters from the original run will be pre-filled, but they can be adjusted as needed before retrying the template.
```

File: docs/assets/software-templates/template-start-over.png
Comment (awanlin) [inline docs/assets/software-templates/template-start-over.png:1]: Can we use screen shots from what the current state of create-app looks like? We should also use the sample template that comes with it as well. The options on the left aren't how most will see Backstages out-of-the-box so we should mimic that as best we can.

File: docs/assets/software-templates/template-task-list.png
Comment (awanlin) [inline docs/assets/software-templates/template-task-list.png:1]: Same here, let's use what comes out-of-the-box with create-app for the images

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this addition to the documentation @subbu-raj!

I've left a few comments, the text itself is very well written we just like to use backticks for code. Then I've left some comments on the images.

**Comment by subbu-raj** [comment]
Hi @awanlin , Thanks for the review and feedback, Can you please review below attached images and let me know, if this is the same image that you are looking for?

**Template Task List Image:**

template-task-list

**Template Start Over Image:**

template-start-over

**Comment by subbu-raj** [comment]
Hi @awanlin , 
For the screenshot images, I have took them by running the Backstage App locally using the instructions from https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#get-started

The options on the left are the ones that I see out of the box, when I launch the app locally. 
Can you let me know, if I should use a different instructions for running the app locally take screenshots from it.

**Comment by awanlin** [comment]
You'll want to use a new Backstage instance created following the guide that Adopters would use to start -  https://backstage.io/docs/getting-started/ - so the results of running: `npx @backstage/create-app@latest`

**Review by awanlin — APPROVED** [review]
Awesome, this looks perfect IMO! 🚀

**Pull request merged**